Source/Urho3D/Network/Transport/DataChannel/DataChannelConnection.cpp

```diff
@@ -67,9 +67,9 @@ bool DataChannelConnection::Connect(const URL& url)
 
 void DataChannelConnection::Disconnect()
 {
+    AddRef(); // Ensure this object is alive until all callbacks are done executing.
```

Comment:
This one seems incorrect to me. `ReleaseRef()` is called in `OnDataChannelDisconnected`, but if we do not have a peer then this callback will not be invoked, leaving us with a dangling reference. `peer_` should always be initialized in a well-formed connection, because it is created in `DataChannelConnection::InitializeFromSocket`, which is either called by a server or by `DataChannelConnection::Connect`. Deeper investigation is needed here, or a sample where i could reproduce the issue and debug it.